doc/source/tune/api_docs/trainable.rst

```diff
@@ -16,6 +16,10 @@ For the sake of example, let's maximize this objective function:
     :start-after: __example_objective_start__
     :end-before: __example_objective_end__
 
+In both APIs, the ``config`` dictionary is populated automatically by Tune and corresponds to the hyperparameters selected for the trial from the :ref:`search space <tune-key-concepts-search-spaces>`.
+
+.. tip:: Avoid passing large objects as hyperparameters in the ``config`` directory, as that will incur a large performance overhead. Use :ref:`tune-with-parameters` to pass large objects in or, if possible, load them inside your function from disk (making sure that all nodes have access to the files) or cloud storage. See :ref:`tune-bottlenecks` for more information.
+
```

Comment:
This was actually quite confusing when reading, since `config` isn't introduced until later in the document. Can we instead have some slight duplication of defining `config` when introducing each API, and centralize the tip, maybe in [Working with Tune Search Spaces](https://docs.ray.io/en/master/tune/tutorials/tune-search-spaces.html)?